core/block.go

```diff
@@ -121,12 +128,50 @@ func VerifyBlockHash(b *Block, network utils.Network) error {
 		}
 
 		if hash, err := blockHash(b, network, overrideSeq); err != nil {
-			return err
+			return []error{err}
 		} else if hash.Equal(b.Hash) {
 			return nil
 		}
 	}
-	return ErrCantVerifyBlockHash
+	return []error{ErrCantVerifyBlockHash}
+}
+
+func verifyTransactionHashes(txs []Transaction, n utils.Network) []error {
```

Comment:
we should benchmark if this is actually faster, we already have `runtime.NumCPU` level of parallelism at block level. If we add parallelism at txn level as well we potentially going to have `runtime.NumCPU ^ 2` number of CPU heavy goroutines in-flight which will lead to starvation.

Reply:
Having tested the sync times it looks like this is worse than sequential verification so I will revert. On `develop` it was around 1 minute faster to sync 429 blocks vs this branch.

Reply:
I will make it sequential

Reply:
done